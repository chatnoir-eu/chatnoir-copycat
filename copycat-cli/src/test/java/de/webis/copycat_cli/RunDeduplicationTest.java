package de.webis.copycat_cli;

import java.nio.file.Files;
import java.nio.file.Path;

import org.approvaltests.Approvals;
import org.junit.Test;

import lombok.SneakyThrows;

public class RunDeduplicationTest {

	@Test
	public void testExceptionIsThrownOnRunWitDuplicatedIds() {
		Approvals.verify(argsForInputFile("run-with-duplicated-ids.txt"));
	}
	
	@Test
	public void approveRunWithoutDuplicatesIsNotChanged() {
		Approvals.verify(argsForInputFile("run-without-duplicates.txt"));
	}

	@Test
	public void approveRunWithDuplicatedTopics() {
		Approvals.verify(argsForInputFile("run-with-duplicated-topics.txt"));
	}
	
	@Test
	public void approveRunWithSingleDuplicate() {
		Approvals.verify(argsForInputFile("run-with-single-duplicate.txt"));
	}
	
	@SneakyThrows
	private static String argsForInputFile(String inputFile) {
		Path outputFile = Files.createTempFile("unit-test-copycat", ".jsonl");
		
		String[] args = new String[] {
			"--outputRunFile", outputFile.toString(),
			"--inputRunFile", "src/test/resources/run-deduplication-test/" + inputFile,
			"--nearDuplicates", "src/test/resources/run-deduplication-test/task1-equivalence-classes-of-near-duplicates-top25.jsonl"
		};
		
		RunDeduplicationApp.deduplicateRun(args);
		String ret = new String(Files.readAllBytes(outputFile));
		outputFile.toFile().delete();
		
		return ret;
	}
}
