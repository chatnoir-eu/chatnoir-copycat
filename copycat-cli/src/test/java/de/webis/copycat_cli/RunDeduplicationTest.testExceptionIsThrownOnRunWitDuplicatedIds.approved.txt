51 Q0 S81a3881b-A1d5a5aa3 0 1.000000 run2021_Baseline_BM25
51 Q0 S885c6b4f-Ad3259e81 1 0.939436 run2021_Baseline_BM25
51 Q0 Saff75a5f-A6bfabf18 2 0.920308 run2021_Baseline_BM25
51 Q0 S6353b9ac-A9cb2062c 3 0.919240 run2021_Baseline_BM25
51 Q0 S13f4e445-A8e79d876 4 0.914850 run2021_Baseline_BM25
51 Q0 S937b1a96-A198bbc54 6 0.910033 run2021_Baseline_BM25
51 Q0 S6353b9ac-A2df6348e 7 0.895392 run2021_Baseline_BM25
51 Q0 Sf27da082-Abdd3fe78 8 0.890464 run2021_Baseline_BM25
51 Q0 Sf27da082-A442fab1b 9 0.873467 run2021_Baseline_BM25
52 Q0 Sb7209cfb-A290d9a97 0 1.000000 run2021_Baseline_BM25
52 Q0 Sc27c8a7b-Add9c0c3c 1 0.983004 run2021_Baseline_BM25
52 Q0 Sc27c8a7b-Ad6bccea5 2 0.914491 run2021_Baseline_BM25
52 Q0 Sb7209cfb-Abebb4467 3 0.906941 run2021_Baseline_BM25
52 Q0 Sc27c8a7b-A17ee9c30 4 0.833355 run2021_Baseline_BM25
52 Q0 Sb7209cfb-Ab75c8938 5 0.825929 run2021_Baseline_BM25
52 Q0 Sb7209cfb-Afd20db15 6 0.806349 run2021_Baseline_BM25
52 Q0 Sc27c8a7b-A84e646ee 7 0.800654 run2021_Baseline_BM25
52 Q0 Sb7209cfb-A8ea181e5 8 0.799047 run2021_Baseline_BM25
52 Q0 Sc27c8a7b-A82679116 9 0.794989 run2021_Baseline_BM25
